hammers/scripts/lease_stack_reaper.py

```diff
@@ -147,22 +147,37 @@ def send_delete_notification(gpu_user, leases, sender):
         html)
 
 
-def collect_user_leases(db):
+def collect_user_leases(auth):
     users = {}
+    users_by_id = keystone.users(auth)
+    hosts_by_id = blazar.hosts(auth)
+    allocs_by_lease = collections.defaultdict(list)
+
+    for alloc in blazar.host_allocations(auth):
+        for r in alloc['reservations']:
+            allocs_by_lease[r['lease_id']].append(alloc['resource_id'])
+
 
-    for row in query.get_advanced_reservations_by_node_type(db):
-        user_id = row['user_id']
+    for lease_id, lease in blazar.leases(auth).items():
+        user_id = lease['user_id']
+        lease['nodes'] = allocs_by_lease[lease_id]
```

Comment:
This is also a good place to update `start_date` and `end_date`. We are calling `parse_datestr` in a lot of places and it's probably more reliable to do this once, when the data is fetched, before processing. The rest of the functions can then assume it's a dateobj.